api/src/main/java/org/apache/iceberg/io/FileIO.java

```diff
@@ -20,7 +20,7 @@
 package org.apache.iceberg.io;
 
 import java.io.Serializable;
-import java.util.Map;
+import org.apache.iceberg.catalog.CatalogConfigurable;
```

Comment:
@rdblue, one thing we need to think about is how we want to pass properties down.  I feel like CatalogConfigurable is a bit of a misnomer here because we should really be able to specify properties at a Catalog level or a table level (e.g. someone may want to configure table specific credentials that differ from the catalog).  

How these properties are accessed and cascade is somewhat unclear.

Reply:
I remember there was a discussion in #1618 , and we came to the conclusion that a FileIO should just be a catalog level thing, which means the credentials used should also be catalog level configuration. Does Iceberg have access control feature at table level?

Reply:
Do you have a link to what you read? And how does it relate to `CatalogConfigurable`?

Reply:
So, there are some features that we might want to control at the table level, but I think that's just going to complicate things.  @rdblue brought up a good point that table level properties are not available on initial load (only catalog properties), so we would need to thing through what cases would really benefit from table level overrides.  We can revisit that later, so I'll resolve.